lib/util/stream_utils.js

```diff
@@ -31,10 +32,17 @@ shaka.util.StreamUtils = class {
    * @param {!Array.<string>} preferredAudioCodecs
    * @param {number} preferredAudioChannelCount
    * @param {!Array.<string>} preferredDecodingAttributes
+   * @param {?shaka.config.CodecSwitchingStrategy} codecSwitchingStrategy
    */
   static chooseCodecsAndFilterManifest(manifest, preferredVideoCodecs,
       preferredAudioCodecs, preferredAudioChannelCount,
-      preferredDecodingAttributes) {
+      preferredDecodingAttributes, codecSwitchingStrategy) {
+    // If enforcing a codec switch strategy skip filtration
+    if (codecSwitchingStrategy && codecSwitchingStrategy !==
+      shaka.config.CodecSwitchingStrategy.DISABLED) {
+      shaka.log.debug('Skipping chooseCodecsAndFilterManifest');
```

Comment:
See my previous comment about the filtering

Reply:
I agree.  I don't think the filtering here is correct, and I think more work is required.

Codec-switching means we don't have to choose one specific codec, but some care should still be taken to consider codec preferences and to remove redundant streams available in multiple codecs.

Reply:
This also breaks the default codec selection when `preferredVideoCodecs` and `preferredAudioCodecs` are provided.